internal/credential/credential.go

```diff
@@ -0,0 +1,15 @@
+package credential
+
+import "oras.land/oras-go/v2/registry/remote/auth"
+
+func Credential(username, password string) auth.Credential {
+	if username == "" {
+		return auth.Credential{
+			RefreshToken: password,
```

Comment:
I'm a bit confused here. Why do we have `password` as `token`?

Reply:
This is a by-design behavior: if user name is provided, then the password is consider as a refresh token, which is a authorization grant from resource owner in the OAuth model.